src/middleware/flash.ml

```diff
@@ -76,3 +56,40 @@ let put_flash category message request =
       failwith message
   in
   outbox := (category, message)::!outbox
+
+let flash_messages inner_handler request =
+  let outbox = ref [] in
+  let request = Dream.with_local storage outbox request in
+  let%lwt response = inner_handler request in
+  let entries = List.rev !outbox in
+  let existing = Dream.cookie flash_cookie request in
+  log.debug (fun log ->
+      let current =
+        flash request
```

Comment:
Shouldn't this display the contents of `outbox`, rather than reading the cookie, which AFAIK is the flash messages returned in the request? If you'd like to display incoming flash messages as I think this is doing, I think it should be done before the call to `inner_handler`.

Reply:
In the issue, @thangngoc89 wrote that it was "frustrating to debug why flash messages aren't rendered". I thought it made sense to log the flash messages from the current request's cookie because those are the ones available for rendering a response. This also appears to match the behavior of the example code in the issue description. 

I've fixed the ordering with `inner_handler`; that was an oversight on my part.

Reply:
The question is, in the original ordering, why read the cookie on the *request* (`flash request`), rather than reading `outbox`? Aren't the messages in the outbox the ones that will be put in the response?

Reply:
Ah okay, I understood you. I got double-negative-confused by myself :) Okay, I now think this is doing what #90 was asking, and the ordering makes sense now.

Reply:
Could you move the debug print all the way to the top of the middleware? It doesn't seem to depend on any of the values bound before it now, and having them bound there makes it seem like it might depend. It's a clean print of the flash messages being returned to the server. It will especially make it clear that the logging does not depend on the rebinding of `request` to one that has the `outbox` attached.